packages/amplify-graphql-model-transformer/src/resources/custom-resource-lambda/custom-resource-handler.ts

```diff
@@ -0,0 +1,440 @@
+    case 'Update':
+      if (!event.PhysicalResourceId) {
+        throw new Error(`Could not find the physical ID for the updated resource`);
+      }
+      console.log('Fetching current table state');
+      const describeTableResult = await ddbClient.describeTable({ TableName: event.PhysicalResourceId }).promise();
+      if (!describeTableResult.Table) {
+        throw new Error(`Could not find ${event.PhysicalResourceId} to update`);
+      }
+      // determine if table needs replacement
+      if (isKeySchemaModified(describeTableResult.Table.KeySchema!, tableDef.KeySchema)) {
+        console.log('Update requires replacement');
+
+        console.log('Deleting the old table');
+        await ddbClient.deleteTable({ TableName: event.PhysicalResourceId }).promise();
+        await retry(
+          async () => await doesTableExist(event.PhysicalResourceId!),
+          (res) => res === false,
+        );
+        console.log(`Table '${event.PhysicalResourceId}' does not exist. Deletion is finished.`);
+
+        const createTableInput = toCreateTableInput(tableDef);
+        const response = await createNewTable(createTableInput);
+        result = {
+          PhysicalResourceId: response.tableName,
+          Data: {
+            TableArn: response.tableArn,
+            TableStreamArn: response.streamArn,
+            TableName: response.tableName,
+          },
+        };
+        log('Returning result', result);
+        return result;
+      }
+      // When normal updates happen
+      const nextGsiUpdate = getNextGSIUpdate(describeTableResult.Table, tableDef);
+      log('Computed next update', nextGsiUpdate);
+      if (!nextGsiUpdate) {
+        result = {
+          PhysicalResourceId: event.PhysicalResourceId,
+          Data: {
+            TableArn: describeTableResult.Table.TableArn,
+            TableStreamArn: describeTableResult.Table.LatestStreamArn,
+            TableName: describeTableResult.Table.TableName,
+          },
+        };
+        return result; // nothing to update
+      }
+
+      // TODO: merge gsi update with other table updates
+
+      const updateTableInput: UpdateTableInput = nextGsiUpdate;
+      log('Merged gsi update with other table updates', updateTableInput);
+
+      console.log('Initiating table update');
+      await ddbClient.updateTable(updateTableInput).promise();
+      result = {
+        PhysicalResourceId: event.PhysicalResourceId,
+        Data: {
+          TableArn: describeTableResult.Table.TableArn,
+          TableStreamArn: describeTableResult.Table.LatestStreamArn,
+          TableName: describeTableResult.Table.TableName,
+        },
+      };
+      return result;
+    case 'Delete':
+      console.log('Initiating table deletion');
+      if (!event.PhysicalResourceId) {
+        throw new Error(`Could not find the physical ID for the resource`);
+      }
+      try {
+        await ddbClient.deleteTable({ TableName: event.PhysicalResourceId }).promise();
```

Comment:
For the table replacement the confirmation of table being deleted is required as the next step is to create the table again with the exact physical id(table name). For the CFN delete event, there is no follow up steps after the deletion so we just kick off the API call here. If the error is thrown during deletion, the CFN will tell user the resource is failed to delete and let them to delete manually. 

The initial thought is to save the cost as the retry listener will take a lot of lambda execution time. This can be mitigated by adding listener in `isComplete`. Mark this as TODO to address in the latter PR.